update chopin posts (#37)
apiobuild/apiobuild.github.io: content/blog/how-to-create-web-store-with-apio.md

  </div>
</div>

Here's an example:

<img src="/images/blog/catalog-sample.png" class="post-img">
<br></br>

Fields are defined as the following:

> - name: unique name/id contains **only dashes, underscores, and alphanumeric characters** for each product

Reviewer: ```suggestion
- name: unique name/id contains **only dashes, underscores, and alphanumeric are allowed** for each product
```

> - nickname: item names that will appear on the store page, they can be **any language and symbol, even emoji**

Reviewer: ```suggestion
- nickname: item names that will appear on the store page, they can be **any language, symbol, and even emoji**
```

- description *(optional)*: more information of the product
- product_url *(optional)*: link to the product website
- image_url: url of the product image
- price: product price
- max_qty: maximum quantity that one customer can buy. **If max_qty = 0, it will show 'Out of Stock'**.
- category *(optional)*: to allow customer to filter products

> *Note: There should not be empty line between rows.*

Reviewer: ```suggestion
*Note: There should not be empty line(s) between rows.*
```

<video width="100%" loop="true" autoplay="true" controls style="align: center">
<source src="/video/copy-paste-catalog-google-sheet.mp4" type="video/mp4" />
</video>

## Step 2: Authorize Catalog Google Sheet on Telescope

Go to [Telescope (apio microservices platform)](https://telescope.apiobuild.com/). Log in into your **existing google account**.

Then go to [Waitress](https://telescope.apiobuild.com/app/waitress) and follow the instruction on the **configure tab** to authorize your google sheet: 
1. Click `Share` at the top right corner of your google sheet
2. Add the Waitress as an editor
3. Copy and paste your google sheer URL. Hit `Authorize`

After authorization, only you can access your sheet through our API. Even us cannot edit your google sheets!

<video width="100%" loop="true" autoplay="true" controls style="align: center">
<source src="/video/authorize-catalog-google-sheet.mp4" type="video/mp4" />
</video>

## Step 3: Create Order Google Sheet

Repeat [Step 1](#step-1-create-catalog-google-sheet) and [Step 2](#step-2-authorize-catalog-google-sheet-on-telescope) to **create another google sheet** with different header. 

> This sheet will be storing **new order** that customers submitted on your store. So just copy and paste the following columns to the sheet, and you don't need to input other information:

Reviewer: ```suggestion
This sheet will be storing **new orders** that customers submitted to your store. Copy and paste the following columns to the sheet, no need to input additional information:
```

  </div>
</div>

*Note: Don't worry if somehow you can't copy everything, just make sure you have **'order_number' in the A1 cell**. The rest will magically appear when an order is placed.*

## Step 4: Create the Store

> After configuring both catalog and order in google sheet, now we can head to [Chopin](https://telescope.apiobuild.com/app/chopin) to setup our store. Enter the following information to the cooresponding fields. You can update this information anytime (except store name and google sheet urls) and the updates will be reflected in real time.

Reviewer: ```suggestion
After configuring both catalog and order in waitress app, we can head to [Chopin](https://telescope.apiobuild.com/app/chopin) to create our store. Enter the following information to the corresponding fields. You can update this information anytime (except store name and google sheet URLs) and the updates will be reflected in real-time.
```

**Customize Your Store**

- Store Name: name of your store
- Logo url *(optional)*: it can even be a gif
- Announcement *(optional)*: it will show at the top of your store. You can update it anytime.
- Store Description *(optional)*: any information you'd like to include. Up to 250 characters.
- Store Background Image url  *(optional)*: add a background image
- Store Background Color  *(optional)*: you can easily select color from swatch or [hex code](https://htmlcolorcodes.com/)
- Contact and Social Plug-ins *(optional)*: currently we support email address, Facebook, Instagram, and Line

**Tax and Shipping Set-up**

> - Tax Rate *(optional)*: if Tax Rate = 0, it will not be shown in the store created

Reviewer: ```suggestion
- Tax Rate *(optional)*: if Tax Rate = 0, it will not be shown on the store page
```

- Order Google Sheet url: copy and paste the authorized order sheet URL from [Step 3](#step-3-create-order-google-sheet)

*Note: In this example, we use [Waitress google sheet API](https://telescope.apiobuild.com/app/waitress) as catalog and order API. We welcome you to bring your own API, <a href="mailto:apiobuild@gmail.com">let us know</a>!*

<video width="100%" loop="true" autoplay="true" controls style="align: center">
<source src="/video/create-store.mp4" type="video/mp4" />
</video>

## Step 5: Start Selling

Voila, the store is now available at: [tinyurl.com/batmans-store](https://tinyurl.com/batmans-store)!

💡You can use services like [tinyurl.com](https://tinyurl.com/) to make the raw url easier to share on facebook, instagram, and among you friends.

💡Any new order will be updated on the order google sheet you created in [Step 3](#step-3-create-order-google-sheet). You can also setup [google sheet notification](https://support.google.com/docs/answer/91588?co=GENIE.Platform%3DDesktop&hl=en) to get real-time updates on new order submission.

Happy “chopin”! 🛍️

# FAQs

> If you see error messages or Patrick Star crying (hopefully really rare), let us trouble-shoot together! <a href="mailto:apiobuild@gmail.com">Contact your favorite developer if you don't find answers here</a>!

Reviewer: ```suggestion
If you see error messages or Patrick Star crying (hopefully really rare), let us troubleshoot together! <a href="mailto:apiobuild@gmail.com">Contact us (your favorite developer) if you don't find answers here</a>!
```

<img src="/images/blog/chopin-error.png" class="post-img">

#### Chopin
  - Go to [Step 2](#step-2-authorize-catalog-google-sheet-on-telescope), check if you authorize both of your google sheets

#### Google Sheet Data
  - In catalog sheet, there should be no empty lines between rows.

>   - In catalog sheet, `name` of each product can not be empty nor can be characters other than numbers or alphabets. There should not be products shared the same `name`.

Reviewer: ```suggestion
  - In the catalog sheet, `name` of each product can not be empty nor can be characters other than numbers or alphabets. Product names must be unique.
```
